src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -678,9 +691,11 @@ internal static bool Dispatch()
                     outerWorkQueue.EnsureThreadRequested();
             }
 
+#if !CORERT
```

Comment:
Instead of this ifdef, it would be better to move:

```
               // If we get here, it's because our quantum expired.  Tell the VM we're returning normally.
                return true;
```

from above to here so that the C# compiler does not complain about unreachable code.

Reply:
This will go away after `Task` is moved to shared.

Reply:
I removed the code blocks completely. May need to bring a little bit of the code back depending on the answer to https://github.com/dotnet/coreclr/pull/22105#discussion_r249820517.